fix RHTAPBUGS-1305 setup git credential helper for private repos

rhopp opened the pull request:
update-deployment was using empty variable ${hostname} and it also wasn't setting up the ~/.gitconfig properly.

In acs-deploy-check, no credentials were being used whatsoever, so the gitops repo clone failed which caused failure of that task.

chmeliik commented:
/ok-to-test

chmeliik commented on task/update-deployment/0.1/update-deployment.yaml:
```diff
@@ -40,11 +40,13 @@ spec:
           else
             origin_with_auth=https://${password}@${remote_without_protocol}.git
           fi
+           echo -e "[credential \"https://${hostname}\"]\n  helper = store" > "${HOME}/.gitconfig"
```
I think this task does auth with this line below
```
git remote set-url origin $origin_with_auth
```
But .git-credentials + .gitconfig does seem like a better approach. Maybe you can remove that line and the `origin_with_auth` variable instead

chmeliik replied:
Ah, and I just noticed that the script doesn't create .git-credentials when there's only a password and no username. That would need a fix too

rhopp replied:
well... I don't know exactly how to do this now - I've tried locally to put GH/Gitlab token to .git-credentials without username and it didn't work. 
Any idea?

chmeliik replied:
Hmm, I see. I feel like I managed to make that work at some point in the past, let me check.

If not, we can just consider that case unsupported

rhopp replied:
Any luck with trying that? Should we merge as-is?

chmeliik replied:
For github, this works:

```bash
echo "https://whatever:$gh_token@github.com/my-org/my-private-repo.git" > ~/.git-credentials
```

But that may be specific to github. Let's just not support the case where the secret only contains a password (except in the update-deployment task, which already supports it so it would be a breaking change).

chmeliik replied:
But in that case, let's align the code with the git-clone and prefetch-dependencies tasks to avoid introducing even more inconsistency than we already have

rhopp commented on task/update-deployment/0.1/update-deployment.yaml:
```diff
@@ -40,11 +40,13 @@ spec:
           if test -f /gitops-auth-secret/username ; then
             username=$(cat /gitops-auth-secret/username)
+            hostname=$(echo $PARAM_GITOPS_REPO_URL | awk -F/ '{print $3}')
             echo "https://${username}:${password})@${hostname}" > "${HOME}/.git-credentials"
```
at some version I tested I fixed this... I have no idea how it got back here... fixing

chmeliik commented:
Just FYI, you can also run checkton locally with `hack/checkton-local.sh` (unless you're on an ARM Mac)

rhopp commented:
@chmeliik I've reworked the PR so the tasks are now using workspaces and same logic as for example git-clone task.
This will also need a change in pipelineruns to add the workspace (https://github.com/redhat-appstudio/tssc-sample-pipelines/blob/main/pac/source-repo/docker-push.yaml#L39). I'll probably need help from somebody. I don't know all the places necessary and if those files are generated/synced from somewhere or I need to change them directly.

This PR now also includes fix for an issue, where the pipelines were not working (`tasks.build-container.results.status` did not exist or something like that.  Removing `results` fixed it.

I've tested this PR by pushing the (kustomize) generated pipeline together with the changed tasks into my repo (https://github.com/rhopp/pipeline-testing/tree/main) and using this in my private gitlab repo (generated by my RHTAP instance) and changed the PipelineRun to contain my changed pipeline and tasks as well as the new workspace:
```
apiVersion: tekton.dev/v1
kind: PipelineRun
metadata:
  name: goapp-gl-private-0829-on-push
  annotations:
    pipelinesascode.tekton.dev/on-event: "[push]"
    pipelinesascode.tekton.dev/on-target-branch: "[main]"
    pipelinesascode.tekton.dev/max-keep-runs: "2"
    pipelinesascode.tekton.dev/pipeline: "https://raw.githubusercontent.com/rhopp/pipeline-testing/main/docker-build-rhtap.yaml"
    pipelinesascode.tekton.dev/task-0: "https://raw.githubusercontent.com/redhat-appstudio/tssc-sample-pipelines/main/pac/tasks/init.yaml"
    pipelinesascode.tekton.dev/task-1: "https://raw.githubusercontent.com/redhat-appstudio/tssc-sample-pipelines/main/pac/tasks/git-clone.yaml"
    pipelinesascode.tekton.dev/task-2: "https://raw.githubusercontent.com/redhat-appstudio/tssc-sample-pipelines/main/pac/tasks/buildah-rhtap.yaml"
    pipelinesascode.tekton.dev/task-3: "https://raw.githubusercontent.com/redhat-appstudio/tssc-sample-pipelines/main/pac/tasks/acs-image-check.yaml"
    pipelinesascode.tekton.dev/task-4: "https://raw.githubusercontent.com/redhat-appstudio/tssc-sample-pipelines/main/pac/tasks/acs-image-scan.yaml"
    pipelinesascode.tekton.dev/task-5: "https://raw.githubusercontent.com/rhopp/pipeline-testing/main/acs-deploy-check.yaml"
    pipelinesascode.tekton.dev/task-6: "https://raw.githubusercontent.com/rhopp/pipeline-testing/main/update-deployment.yaml"
    pipelinesascode.tekton.dev/task-7: "https://raw.githubusercontent.com/redhat-appstudio/tssc-sample-pipelines/main/pac/tasks/show-sbom-rhdh.yaml"
    pipelinesascode.tekton.dev/task-8: "https://raw.githubusercontent.com/redhat-appstudio/tssc-sample-pipelines/main/pac/tasks/summary.yaml"
  labels:
    argocd/app-name: goapp-gl-private-0829
    janus-idp.io/tekton: goapp-gl-private-0829
    backstage.io/kubernetes-id: goapp-gl-private-0829
    backstage.io/kubernetes-namespace: rhtap-app
    app.kubernetes.io/part-of: goapp-gl-private-0829
spec:
  params:
    - name: dockerfile
      value: docker/Dockerfile
    - name: git-url
      value: '{{repo_url}}'
    - name: image-expires-after
      value: 5d
    - name: output-image
      value: rhtap-quay-quay-rhtap-quay.apps.rhopp240829.devcluster.openshift.com/rhtap/goapp-gl-private-0829:{{revision}}
    - name: path-context
      value: .
    - name: revision
      value: '{{revision}}'
    - name: event-type
      value: '{{event_type}}'
    - name: gitops-auth-secret-name
      value: gitlab-auth-secret
  pipelineRef:
    name: docker-build-rhtap
  workspaces:
    - name: git-auth
      secret:
        secretName: "{{ git_auth_secret }}"
    - name: gitops-auth
      secret:
        secretName: $(params.gitops-auth-secret-name)
    - name: workspace
      volumeClaimTemplate:
        spec:
          accessModes:
            - ReadWriteOnce
          resources:
            requests:
              storage: 1Gi
```
